OpenRA.Game/FileSystem/FileSystem.cs

```diff
@@ -103,63 +85,63 @@ public void Mount(string name, string annotation = null)
 				a();
 		}
 
-		void MountInner(IReadOnlyPackage package)
+		public void Mount(IReadOnlyPackage package)
 		{
-			MountedPackages.Add(package);
-
-			foreach (var hash in package.ClassicHashes())
+			var mountCount = 0;
+			if (mountedPackages.TryGetValue(package, out mountCount))
 			{
-				var packageList = classicHashIndex[hash];
-				if (!packageList.Contains(package))
-					packageList.Add(package);
+				mountedPackages[package] = mountCount + 1;
```

Comment:
This will also need to enumerate the package contents and bump up the per-file pacakge priority if it is not already at the end of the list.